website/content/docs/waypoint-hcl/variables/input.mdx

```diff
@@ -0,0 +1,353 @@
+---
+layout: docs
+page_title: Input Variables - Variables - waypoint.hcl
+description: |-
+  Input variables allow you to customize Waypoint configuration according to 
+  set parameters. This section documents input variable syntax, including how to
+  define, reference, and use input variables.
+---
+
+# Input Variables
+
+Input variables let you customize aspects of Waypoint configuration with values
+that may change, or that need to be set per Waypoint run.
+
+You can parameterize your configuration by declaring variables in the
+waypoint.hcl, and setting their values using CLI options, environment variables,
+`auto` files, and the Waypoint server UI.
+
+> **Hands-on:** Try the [Input Variables]() tutorial on HashiCorp Learn.
+
+## Declaring an Input Variable
+
+Each input variable referenced in the waypoint.hcl must be declared using a
+`variable` block:
+
+```hcl
+variable "port" {
+  type = number
+  default = 8080
+}
+
+variable "image_name" {
+  type    = string
+  default = "latest"
+}
+
+variable "docker_labels" {
+  type = object({
+    environment = string
+    log_level   = string
+  })
+  default =  {
+      environment = test
+      log_level   = debug
+    }
+}
+```
+
+The label after the `variable` keyword is a name for the variable, which must
+be unique among all variables in the same waypoint.hcl. This name is used to
+assign a value to the variable and to reference the variable's value when
+evaluating the configuration.
+
+The name of a variable can be any valid [identifier](/docs/waypoint-hcl/syntax#identifiers).
+
+## Arguments
```

Comment:
You may not need this with the stanza reference docs... up to you. There'll be some duplication either way.